### ISSUE TITLE
[module] Financial Planning

### ISSUE THREAD

**Issue by akalitenya** [so-epic/kickoff#8]
The main objective of Financial Planning is to answer the question of What If.

EpicML references data tables prepared by DBT as a source (or other tables in data warehouse).

**Comment by bebelabs** [comment]
Sorry, non-technical co-founder here. What is DBT?  Do you mean EpicML references data tables prepared as a result of financial planning?

**Comment by akalitenya** [comment]
Steps

1.
- There are many tables with raw data in data warehouse
- These tables are periodically updated with more recent data from sources
- We periodically run processes that transform these tables into other more useful tables (we use DBT for this)

2.
- Data from Transfromed tables are used to generate the FP tables (Financial Planning)
- User can view and edit some values in FP table (edit financial "drivers") 
- changes of 1 cell in FP table can cause chain reactions of changes in same FP table and in other "connected" FP tables (like in Excel/Spreadsheets)
3.
- we update changed FP tables in data warehouse
- We can also periodically run processes that transform data from initial Transformed tables and FP tables to produce new useful tables (and we can use DBT for this)
- data from Transformed Tables and from FP tables can be referenced and used by BI module of Epic app (or any other BI tool) 

All transformation steps can be represented as Direct Acyclyc Graph nodes.

The steps in Level 2 are more likely to be performed outside the SQL data warehouse, but their result will be added to the data warehouse.

Thus, the FP module uses data from Transformed tables and produces new FP tables that can be used in next steps of transformations (using DBT) or quered by BI module

**Comment by bebelabs** [comment]
Interesting article that breaks down the components of financial planning https://www.digitalistmag.com/finance/2018/09/27/what-finance-teams-should-look-for-in-intelligent-ifpa-solution-06186629

**Comment by PVidalArean** [comment]
> The main objective of Financial Planning is to answer the question of What If.

@akalitenya that is the definition of scenario planning, not financial planning. Financial planning responds to what I need to achieve my objective.

**Comment by akalitenya** [comment]
Thank you @PVidalArean , ideally we need to create a glossary of terms. Under the term "Financial Planning", I call the module of Epic product where we can calculate the Metrics (another term) based on actual data and calculate the future state of these Metrics. The main thing is that we do not get confused with each other using different terms.

**Comment by PVidalArean** [comment]
Makes sense. Just keep that in mind in case you prefer to give it a different term or just keep it understanding that is more “scenario planning” that “financial planning”. 
Whatever works for you guys.

**Comment by bebelabs** [comment]
Yes Alex, great call! This space can get very confusing very fast because the concepts are very closely related and interdependent.  I just started a glossary locally.  Because you have such clarity of thought and ability to communicate it, want to kick off the glossary and I can add mine to it?
